hail/src/main/scala/is/hail/types/physical/PContainer.scala

```diff
@@ -89,7 +89,9 @@ abstract class PContainer extends PIterable {
   def nextElementAddress(currentOffset: Code[Long]): Code[Long]
 }
 
-abstract class PIndexableValue extends PValue with SIndexableValue
+abstract class PIndexableValue extends PValue with SIndexableValue {
+  def pt: PContainer
```

Comment:
Where is this used? We should be moving away from using ptypes on PCodes/PValues in favor of STypes or methods on the codes/values themselves

Reply:
https://github.com/hail-is/hail/pull/9824/files#diff-5fcfe4833387843730db9b875d97b36742722849a88b4685797bed3b19a66ef5R2819

I just did this so I could call `elementType` on something I knew to be a `PArray` so I could call `nDims` on `PNDArray` type contained in the array.

Reply:
cool, thanks. I realized today that stypes are missing the virtual types. If SNDArray had `vt: TNDArray` you could get the ndims right?

Reply:
If we put `elementType` on `PIndexableCode` that would remove the need for this.

Reply:
awesome. I can make that PR tomorrow, does that work?

Reply:
I definitely want to be able to get the number of dimensions easily if I'm holding an `SNDArray`. Whether that's by going through the virtual type or the `SNDArray` just has its own `nDims` field on it, definitely should be an option. 

Separately though, I want to be able to make a decision based on the element type of a regular array. I could have just done:

```
coerce[PArray](myArrayPValue.pt).elementType
```

but I just made this change to avoid the cast. Any way you would like me to get the number of dimensions of the ndarray type which is contained in this array is fine with me.

Reply:
I can remove this change and do what I want without any other changes, but I still think it would be helpful in future to be able to get element type of arrays.